src/Umbraco.Web/PublishedCache/NuCache/NuCacheComposer.cs

```diff
@@ -14,36 +15,22 @@ public override void Compose(Composition composition)
             base.Compose(composition);
 
             var serializer = ConfigurationManager.AppSettings["Umbraco.Web.PublishedCache.NuCache.Serializer"];
-            composition.Register<INuCachePropertyOptionsFactory, AppSettingsNuCachePropertyMapFactory>();
-            composition.Register<Lz4DictionaryOfPropertyDataSerializer, Lz4DictionaryOfPropertyDataSerializer>();
-
-            // TODO: Based on our findings it seems like this should not be configurable, we should just be using this because it's better
-            if (serializer == "MsgPack")
+            if (serializer != "MsgPack")
```

Comment:
By default it's not messagepack, unless a migration adds the appsetting

Reply:
Yep, we'll most likely ship with this setting which means it will be opt-in so turned on for new sites but for upgrades people will have to opt in manually. I'll add this to the TODO notes on this issue (which will keep growing). Possibly we make it default for everything including upgrades which will then require migrations.